From b9461bd4f7d26046e113cff955b3e5e88873b10b Mon Sep 17 00:00:00 2001
From: Kalyan Kondapally <kalyan.kondapally@intel.com>
Date: Tue, 23 Dec 2014 04:38:36 -0800
Subject: [PATCH 02/14] Remove usage of DesktopFactory.

This is highly experimental. Eventually, the goal is to only have
platform specific parts in Wayland and core implementation as part
of Chromium. Once, we have more matured implementation, initiate
discussion upstream.
---
 ui/views/views.gyp                                 |  9 ----
 .../widget/desktop_aura/desktop_factory_ozone.cc   | 29 -------------
 .../widget/desktop_aura/desktop_factory_ozone.h    | 50 ----------------------
 .../widget/desktop_aura/desktop_screen_ozone.cc    | 15 -------
 .../desktop_aura/desktop_window_tree_host_ozone.cc | 20 ---------
 5 files changed, 123 deletions(-)
 delete mode 100644 ui/views/widget/desktop_aura/desktop_factory_ozone.cc
 delete mode 100644 ui/views/widget/desktop_aura/desktop_factory_ozone.h
 delete mode 100644 ui/views/widget/desktop_aura/desktop_screen_ozone.cc
 delete mode 100644 ui/views/widget/desktop_aura/desktop_window_tree_host_ozone.cc

diff --git a/ui/views/views.gyp b/ui/views/views.gyp
index 6da0fa7..2ba2a8d 100644
--- a/ui/views/views.gyp
+++ b/ui/views/views.gyp
@@ -460,12 +460,6 @@
       'widget/desktop_aura/desktop_window_tree_host_win.cc',
       'widget/desktop_aura/desktop_window_tree_host_win.h',
     ],
-    'views_desktop_aura_ozone_sources': [
-      'widget/desktop_aura/desktop_factory_ozone.cc',
-      'widget/desktop_aura/desktop_factory_ozone.h',
-      'widget/desktop_aura/desktop_screen_ozone.cc',
-      'widget/desktop_aura/desktop_window_tree_host_ozone.cc',
-    ],
     'views_test_support_sources': [
       'controls/textfield/textfield_test_api.cc',
       'controls/textfield/textfield_test_api.h',
@@ -726,9 +720,6 @@
             ['OS == "win"', {
               'sources': [ '<@(views_desktop_aura_win_sources)' ],
             }],
-            ['use_ozone==1', {
-              'sources': [ '<@(views_desktop_aura_ozone_sources)' ],
-            }],
           ],
         }],
         ['OS=="mac"', {
diff --git a/ui/views/widget/desktop_aura/desktop_factory_ozone.cc b/ui/views/widget/desktop_aura/desktop_factory_ozone.cc
deleted file mode 100644
index e0a4489..0000000
--- a/ui/views/widget/desktop_aura/desktop_factory_ozone.cc
+++ /dev/null
@@ -1,29 +0,0 @@
-// Copyright 2013 The Chromium Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-#include "ui/views/widget/desktop_aura/desktop_factory_ozone.h"
-
-#include "base/logging.h"
-
-namespace views {
-
-// static
-DesktopFactoryOzone* DesktopFactoryOzone::impl_ = NULL;
-
-DesktopFactoryOzone::DesktopFactoryOzone() {
-}
-
-DesktopFactoryOzone::~DesktopFactoryOzone() {
-}
-
-DesktopFactoryOzone* DesktopFactoryOzone::GetInstance() {
-  CHECK(impl_) << "DesktopFactoryOzone accessed before constructed";
-  return impl_;
-}
-
-void DesktopFactoryOzone::SetInstance(DesktopFactoryOzone* impl) {
-  impl_ = impl;
-}
-
-} // namespace views
diff --git a/ui/views/widget/desktop_aura/desktop_factory_ozone.h b/ui/views/widget/desktop_aura/desktop_factory_ozone.h
deleted file mode 100644
index 2af191b..0000000
--- a/ui/views/widget/desktop_aura/desktop_factory_ozone.h
+++ /dev/null
@@ -1,50 +0,0 @@
-// Copyright 2013 The Chromium Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-#ifndef UI_VIEWS_WIDGET_DESKTOP_AURA_DESKTOP_FACTORY_OZONE_H_
-#define UI_VIEWS_WIDGET_DESKTOP_AURA_DESKTOP_FACTORY_OZONE_H_
-
-#include "ui/views/views_export.h"
-
-namespace gfx {
-class Rect;
-class Screen;
-}
-
-namespace views {
-class DesktopNativeWidgetAura;
-class DesktopWindowTreeHost;
-
-namespace internal {
-class NativeWidgetDelegate;
-}
-
-class VIEWS_EXPORT DesktopFactoryOzone {
- public:
-  DesktopFactoryOzone();
-  virtual ~DesktopFactoryOzone();
-
-  // Returns the instance.
-  static DesktopFactoryOzone* GetInstance();
-
-  // Sets the implementation delegate. Ownership is retained by the caller.
-  static void SetInstance(DesktopFactoryOzone* impl);
-
-  // Delegates implementation of DesktopWindowTreeHost::Create externally to
-  // Ozone implementation.
-  virtual DesktopWindowTreeHost* CreateWindowTreeHost(
-      internal::NativeWidgetDelegate* native_widget_delegate,
-      DesktopNativeWidgetAura* desktop_native_widget_aura) = 0;
-
-  // Delegates implementation of DesktopScreen externally to
-  // Ozone implementation.
-  virtual gfx::Screen* CreateDesktopScreen() = 0;
-
- private:
-  static DesktopFactoryOzone* impl_; // not owned
-};
-
-}  // namespace views
-
-#endif // UI_VIEWS_WIDGET_DESKTOP_AURA_DESKTOP_FACTORY_OZONE_H_
diff --git a/ui/views/widget/desktop_aura/desktop_screen_ozone.cc b/ui/views/widget/desktop_aura/desktop_screen_ozone.cc
deleted file mode 100644
index 9708ad8..0000000
--- a/ui/views/widget/desktop_aura/desktop_screen_ozone.cc
+++ /dev/null
@@ -1,15 +0,0 @@
-// Copyright 2013 The Chromium Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-#include "ui/views/widget/desktop_aura/desktop_screen.h"
-
-#include "ui/views/widget/desktop_aura/desktop_factory_ozone.h"
-
-namespace views {
-
-gfx::Screen* CreateDesktopScreen() {
-  return DesktopFactoryOzone::GetInstance()->CreateDesktopScreen();
-}
-
-}  // namespace views
diff --git a/ui/views/widget/desktop_aura/desktop_window_tree_host_ozone.cc b/ui/views/widget/desktop_aura/desktop_window_tree_host_ozone.cc
deleted file mode 100644
index 1a78a9a..0000000
--- a/ui/views/widget/desktop_aura/desktop_window_tree_host_ozone.cc
+++ /dev/null
@@ -1,20 +0,0 @@
-// Copyright 2013 The Chromium Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-#include "ui/aura/window_tree_host.h"
-#include "ui/views/widget/desktop_aura/desktop_factory_ozone.h"
-#include "ui/views/widget/desktop_aura/desktop_window_tree_host.h"
-
-namespace views {
-
-DesktopWindowTreeHost* DesktopWindowTreeHost::Create(
-    internal::NativeWidgetDelegate* native_widget_delegate,
-    DesktopNativeWidgetAura* desktop_native_widget_aura) {
-  DesktopFactoryOzone* d_factory = DesktopFactoryOzone::GetInstance();
-
-  return d_factory->CreateWindowTreeHost(native_widget_delegate,
-                                         desktop_native_widget_aura);
-}
-
-}  // namespace views
-- 
1.9.1

